### DISCUSSION TITLE
Global logger that is not limited to integrations

### DISCUSSION THREAD

**Discussion by idlist** [withastro/roadmap#693] [Proposal]
### Body

# Summary

Expose Astro's internal logger to global scope so that it can be used in other scripts or components.

# Background & Motivation

Had written some custom i18n utilities inside the `src` folder, and want some logging as hint if I'm making some mistake. 

# Goals

The background is a more specific example of the situation where the user want to intentionally keep some logging code in their code base, rather than using `console.log` as a temporary debug print.

Also, the output of `console.log` would be (somewhat falsely) cleared by Astro's logger if Astro's logger are logging the same content (the `(xN)` mark) while `console.log` isn't. I guess there's no easy solution to this issue, and it actually doesn't matter much when using logging as a debug method. but it would be hindering when a constant output is expected.

Something like `import { useLogger } from Astro` inside a `.js/ts` file or `Astro.useLogger` inside an `.astro` file would be useful in this situation.
 
# Example

```ts
import { useLogger } from 'astro'
const logger = useLogger('i18n')
logger.warn('path '${path}' does not exists in the i18n dictionary.')
// output: XX:XX:XX [i18n] path 'XXX' does not exists in the i18n dictionary.
```

**Comment by Fryuni** [comment]
I made this as an integration: https://inox-tools.vercel.app/runtime-logger#for-projects

You can run `astro add @inox-tools/runtime-logger` to get Astro's logger at runtime.

**Reply by KC900201** [reply]
Hi @Fryuni,

I tried to implement the runtime-logger according to the documentation in my project, but I can't implement the runtime logger in the component files due to the error below:

![image](https://github.com/user-attachments/assets/111fb5ef-2313-4244-8ea7-ed50c032b597)

Cause: library _@it-astro:logger_ cannot be found.

I did integrate the logger in the define Config as below in the astro.config.mjs


```javascript
import { defineConfig } from "astro/config"

import { inoxToolsRuntimeLogger } from "@inox-tools/runtime-logger"

export default defineConfig({
  site: "https://astro-devpedia.pages.dev/",
  // Reference - https://inox-tools.fryuni.dev/runtime-logger
  integrations: [inoxToolsRuntimeLogger()]
})
```
May you suggest the solution for this?

**Reply by Fryuni** [reply]
There shouldn't be `{}` surrounding the import in the config. It's a default export.
The named export is for other integrations to use

**Reply by KC900201** [reply]
Hi @Fryuni,

I followed your advice to remove the parentheses `{}` from the config import. However, the issue persists.

![image](https://github.com/user-attachments/assets/d86f5752-ed77-41c0-a91c-274c46fdf009)

Error screenshot:

![image](https://github.com/user-attachments/assets/2ebd4d07-499c-416a-8339-59e805e7120f)

Any advice on the solution would be much appreciated.

**Reply by ematipico** [reply]
Have you tried to use it? I don't see any code. That's just a warning saying it can't find the types for the virtual module.

If the problem persists, consider filing a bug in the repository of the integration.

**Reply by Fryuni** [reply]
Oh yeah, good catch ema. Types won't be loaded when using a JS config.
But @KC900201 if you are facing any problems on the lib pls open an issue on https://github.com/Fryuni/inox-tools
